src/mode/folding/yaml.js

```diff
@@ -0,0 +1,108 @@
+"use strict";
+
+var oop = require("../../lib/oop");
+var CoffeeFoldMode = require("./coffee").FoldMode;
+var Range = require("../../range").Range;
+
+var FoldMode = exports.FoldMode = function() {};
+oop.inherits(FoldMode, CoffeeFoldMode);
```

Comment:
Since we inherit the CoffeeFoldMode, do we inherit its new `commentBlock`? Could it cause issues given CoffeeScript and YAML multi comments are different? 
e.g. 
```
###
This block 
comment
line
###
```
is not a valid block comment for YAML.

Reply:
It seems to work as desired:

no folding
<img width="508" alt="Screenshot 2023-10-30 at 10 48 19" src="https://github.com/ajaxorg/ace/assets/34573235/1e8743bd-556d-4da1-bc48-61c0347abd95">

folding
<img width="501" alt="Screenshot 2023-10-30 at 10 48 25" src="https://github.com/ajaxorg/ace/assets/34573235/c2e8a036-17a0-4d50-8f4c-a19654b28fdf">